### PR TITLE
[RHCLOUD-35597] create endpoint to event type link from endpoint creation

### PR DESCRIPTION
**Pull request by g-duval**

### CODE REVIEW

File: backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java [hunk 1]
```diff
@@ -371,4 +371,18 @@ public boolean existsByUuidAndOrgId(final UUID endpointUuid, final String orgId)
...
+            Endpoint endpoint = entityManager.createQuery(query, Endpoint.class)
+                .setParameter("id", id)
+                .setParameter("orgId", orgId)
+                .getSingleResult();
+            loadProperties(endpoint);
+            return endpoint;
+        } catch (NoResultException e) {
+            return null;
+        }
+    }
```
Comment (MikelAlejoBR) [inline backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java:387]: I would suggest returning `Optional<Endpoint>` here too!

**Review by MikelAlejoBR — COMMENTED** [review]
LGTM! Great work @g-duval! I'm happy to see all these changes :tada:

File: backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java [hunk 2]
```diff
@@ -466,6 +537,22 @@ protected Endpoint internalCreateEndpoint(
...
+                if (eventTypeFromDb.isPresent()) {
+                    linkedEventTypes.add(eventTypeFromDb.get());
+                } else {
+                    throw new NotFoundException(String.format("Event type '%s' not found", evtId));
+                }
+            }
+            return linkedEventTypes;
+        }
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (g-duval) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java]: You are right, thank you.
I refactored it a bit, mostly based on your suggestion :)

**Review by MikelAlejoBR — APPROVED** [review]

File: backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (MikelAlejoBR) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java]: Looking good!

**Pull request merged**